examples/ai-core/src/e2e/fireworks.test.ts

```diff
@@ -1,398 +1,33 @@
-  });
-
-  describe.each(MODEL_VARIANTS.completion)('Completion Model: %s', modelId => {
-    const model = provider(modelId);
-
-    it('should generate text', async () => {
-      const result = await generateText({
-        model,
-        prompt: 'Complete this code: function fibonacci(n) {',
-      });
-
-      expect(result.text).toBeTruthy();
-      expect(result.usage?.totalTokens).toBeGreaterThan(0);
-    });
-
-    it('should stream text', async () => {
-      const result = streamText({
-        model,
-        prompt: 'Write a Python function that sorts a list:',
-      });
-
-      const chunks: string[] = [];
-      for await (const chunk of result.textStream) {
-        chunks.push(chunk);
-      }
-
-      expect(chunks.length).toBeGreaterThan(0);
-      expect((await result.usage)?.totalTokens).toBeGreaterThan(0);
-    });
-
-    it('should throw error on generate text attempt with invalid model ID', async () => {
-      const invalidModel = provider('no-such-model');
-
-      try {
-        await generateText({
-          model: invalidModel,
-          prompt: 'This should fail',
-        });
-        // If we reach here, the test should fail
-        expect(true).toBe(false);
-      } catch (error) {
-        expect(error).toBeInstanceOf(APICallError);
-        expect(((error as APICallError).data as FireworksErrorData).error).toBe(
-          'Model not found, inaccessible, and/or not deployed',
-        );
-      }
-    });
-
-    it('should throw error on stream text attempt with invalid model ID', async () => {
-      const invalidModel = provider('no-such-model');
-
-      try {
-        const result = streamText({
-          model: invalidModel,
-          prompt: 'This should fail',
-        });
-
-        // Try to consume the stream to trigger the error
-        for await (const _ of result.textStream) {
-          // Do nothing with the chunks
-        }
-
-        // If we reach here, the test should fail
-        expect(true).toBe(false);
-      } catch (error) {
-        expect(error).toBeInstanceOf(APICallError);
-        expect(((error as APICallError).data as FireworksErrorData).error).toBe(
-          'Model not found, inaccessible, and/or not deployed',
-        );
-      }
-    });
-  });
-});
+import { createFeatureTestSuite } from './feature-test-suite';
+
+createFeatureTestSuite({
+  name: 'Fireworks',
+  createChatModelFn: provider.chatModel,
+  createCompletionModelFn: provider.completionModel,
+  createEmbeddingModelFn: provider.textEmbeddingModel,
+  models: {
+    chat: [
+      'accounts/fireworks/models/firefunction-v2',
+      'accounts/fireworks/models/llama-v3p3-70b-instruct',
+      'accounts/fireworks/models/mixtral-8x7b-instruct',
+      'accounts/fireworks/models/qwen2p5-72b-instruct',
+    ],
+    completion: [
+      'accounts/fireworks/models/llama-v3-8b-instruct',
+      'accounts/fireworks/models/llama-v2-34b-code',
+    ],
+    embedding: ['nomic-ai/nomic-embed-text-v1.5'],
+  },
```

Comment:
what are your thoughts about inlining the model construction? it could lead to some duplication here (ok imo), but it would allow us to test wrapped (middleware) models and models with settings as well:

```
models: {
    chat: [
      fireworks('accounts/fireworks/models/firefunction-v2'),
      ...
```

Reply:
(imo this might be a better level of abstraction for test call interface, making it more flexible, but I may be missing some details)